### ISSUE TITLE
Using Term causes the consumer test to fail if self.assertEqual is used

### ISSUE THREAD

**Issue by antweiss** [pact-foundation/pact-python#52]
My code:
```
   expected = {
      u'name':u'#3455',
      u'timestamp': Term('\d+-\d+-\d+T\d+:\d+:\d+.+', u'2017-09-17T19:43:12+03:00'),
    }

    (pact
     .given('build 3455 exists')
     .upon_receiving('a request for build 3455')
     .with_request('get', '/builds/3455')
     .will_respond_with(200, body=expected))


    with pact:
      result = client.build(3455)

    self.assertEqual(result, expected)
```

causes:

FAIL: test_get_build (build_test.GetBuildInfoContract)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "build_test.py", line 38, in test_get_build
    self.assertEqual(result, expected)
AssertionError: {u'timestamp': u'2017-09-17T19:43:12+03:00', u'name': u'#3455'} != {u'timestamp': <pact.matchers.Term object at 0x1019355d0>, u'name': u'#3455'}
- {u'name': u'#3455', u'timestamp': u'2017-09-17T19:43:12+03:00'}
+ {u'name': u'#3455', u'timestamp': <pact.matchers.Term object at 0x1019355d0>}

----------------------------------------------------------------------
Ran 1 test in 0.035s

FAILED (failures=1)

the pact file is written ok but I'd expect this to succeed

**Comment by antweiss** [comment]
By lloking more into this - it looks like a routine for converting the expected body to json should be exported.

**Comment by jawu** [comment]
Would a helper function be a usable solution?
Something like 
`get_generated_values(dict)`
Every value of a key that is an instance of Matcher gets resolved to its generate property and than this could be used for assertion.

**Comment by matthewbalvanz-wf** [comment]
I think created a helper function is certainly an option, but I am interested in how it might be used. @antweiss In your example you are matching a timestamp, and one of the common reasons to use a matcher with a timestamp is automatically generated data. In those cases the helper function likely won't work the way people expect if we return the generated values (the second argument to `Term`), as the consumer code they are testing would usually generate a different timestamp value.

**Comment by jawu** [comment]
In the example above you could use 
`get_generated_values(expected)`
which will result in 
`expected = {
      u'name':u'#3455',
      u'timestamp': u'2017-09-17T19:43:12+03:00'),
    }`
This will also be the response body from the mock service, because it uses the generate value as well.
But a question at the side: we do only test the functionality of the mock service with this assertion, aren't we?

**Comment by matthewbalvanz-wf** [comment]
Looking at this again this morning, I think I was looking at it backward and was thinking these values were in the consumer request side in the example. I have since corrected my thinking, and realize were generating the responses from the mocked service now. A function like `get_generated_values` seems like a good approach here.

@jawu You are correct, this example is just testing the behavior of the mock service, as it does not include any logic from an API layer that might exist in an application.

**Comment by jawu** [comment]
I've added the function to the matchers.py file and opened a pull request.

**Comment by matthewbalvanz-wf** [comment]
#69 was released in v0.13.0, and adds a function to get the generated values from an object that can contain matchers like `Term`, `Like`, etc. The original example would take advantage of it during the assertion:

```python
from pact.matchers import get_generated_values
self.assertEqual(result, get_generated_values(expected))
```
